Make minor language improvements and correct typo (#17338)
mdn/content: files/en-us/web/api/channel_messaging_api/using_channel_messaging/index.md

---
{{DefaultAPISidebar("Channel Messaging API")}}

> The [Channel Messaging API](/en-US/docs/Web/API/Channel_Messaging_API) allows two separate scripts running in different browsing contexts attached to the same document (e.g. two IFrames, or the main document and an IFrame, or two documents via a {{domxref("SharedWorker")}}) to communicate directly, passing messages between each other through two-way channels (or pipes) with a port at each end.

Reviewer: The comma is mandatory, per our [style guide](https://developer.mozilla.org/en-US/docs/MDN/Guidelines/Writing_style_guide#latin_abbreviations).

Also iframe are element names, let's style them as such:
```suggestion
The [Channel Messaging API](/en-US/docs/Web/API/Channel_Messaging_API) allows two separate scripts running in different browsing contexts attached to the same document (e.g., two {{HTMLElement("iframe")}} elements, the main document and a single {{HTMLElement("iframe")}}, or two documents via a {{domxref("SharedWorker")}}) to communicate directly, passing messages between each other through two-way channels (or pipes) with a port at each end.
```

Author: Thanks, will read the style guide next.

In this article we'll explore the basics of using this technology.

{{AvailableInWorkers}}

## Use cases

> Channel messaging is mainly useful in cases where you've got a social site that embeds capabilities from other sites into its main interface via iframes, such as games, address book, or an audio player with personalized music choices. When these act as standalone units, things are OK, but the difficulty comes when you want interaction between the main site and the IFrames, or between different IFrames. For example, what if you wanted to add a contact to the address book from the main site, add high scores from your game to your main profile, or add new background music choices from the audio player to the game? Such things are not so easy using conventional web technology, because of the security models the web uses. You have to think about whether the origins trust each other, and how the messages are passed.

Reviewer: Fix grammar and style `<iframe>`
```suggestion
Channel messaging is mainly useful in cases where you've got a social site that embeds capabilities from other sites into its main interface via iframes, such as games, address books or an audio player with personalized music choices. When these act as standalone units, things are OK, but the difficulty comes when you want interaction between the main site and the {{HTMLElement("iframe")}} elements, or between different {{HTMLElement("iframe")}} elements. For example, what if you wanted to add a contact to the address book from the main site, add high scores from your game to your main profile, or add new background music choices from the audio player to the game? Such things are not so easy using conventional web technology, because of the security models the web uses. You have to think about whether the origins trust each other, and how the messages are passed.
```

## Receiving the port and message in the IFrame

> In the IFrame, we have the following JavaScript:

Reviewer: ```suggestion
In the {{HTMLElement("iframe")}} elements, we have the following JavaScript:
```
